src/Cassandra.Tests/RequestExecutionTests.cs

```diff
@@ -0,0 +1,184 @@
+﻿//
+//       Copyright DataStax, Inc.
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+//
+
+using System;
+using System.Collections.Generic;
+using System.Net;
+
+using Cassandra.Requests;
+using Cassandra.Responses;
+using Cassandra.Serialization;
+
+using Moq;
+
+using NUnit.Framework;
+
+namespace Cassandra.Tests
+{
+    [TestFixture]
+    public class RequestExecutionTests
+    {
+        [Test, TestCase(true), TestCase(false)]
+        public void Should_ThrowException_When_NoValidHosts(bool currentHostRetry)
```

Comment:
`RequestExecution` can be unit tested now, nice!

Reply:
🎉